client.js

```diff
@@ -90,6 +95,18 @@ RingpopClient.prototype._request = function _request(host, endpoint, head, body,
             return;
         }
 
+        if (head && head.gzip === true) {
```

Comment:
`head` in this context is the header of the request, not of the response. In the case you are making a gzipped request to an older version (without gzip support) it will send you the plain information back.

I do think it is also good to signal in the request header that you want the response to be gzipped, but lets confirm in the response header that it is actually gzipped before trying to unzip it. Which involves adding that header I believe: https://github.com/uber/ringpop-node/blob/syncer/server/protocol/sync.js#L63